shell/common/gin_helper/event_emitter_caller.h

```diff
@@ -34,7 +35,15 @@ v8::Local<v8::Value> EmitEvent(v8::Isolate* isolate,
   internal::ValueVector concatenated_args = {gin::StringToV8(isolate, name)};
   concatenated_args.reserve(1 + args.size());
   concatenated_args.insert(concatenated_args.end(), args.begin(), args.end());
-  return internal::CallMethodWithArgs(isolate, obj, "emit", &concatenated_args);
+
+  v8::TryCatch try_catch(isolate);
+  v8::Local<v8::Value> value =
+      internal::CallMethodWithArgs(isolate, obj, "emit", &concatenated_args);
+  if (try_catch.HasCaught()) {
+    LOG(ERROR) << "An unknown error occurred in the event emitter";
```

Comment:
We need to rethrow the caught exception here, we should either send this to the global error handler `global.onerror` or do `Emit('error', exception)` which is more nodejs-esque. 🤔 The NodeJS `error` event handler will dispatch the error event to the global handler for us iirc.

Reply:
Thanks! Will make that change 🙂